rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -444,14 +444,18 @@ BtcBlockStoreWithCache getBtcBlockStore() {
     protected PegTxType getTransactionType(BtcTransaction btcTx) {
         Coin minimumPeginTxValue = bridgeConstants.getMinimumPeginTxValue(activations);
         Script retiredFederationP2SHScript = provider.getLastRetiredFederationP2SHScript().orElse(null);
-        Address oldFederation = Address.fromBase58(bridgeConstants.getBtcParams(), bridgeConstants.getOldFederationAddress());;
+
+        Address oldFederationAddress = null;
+        if (activations.isActive(ConsensusRule.RSKIP199)) {
+            oldFederationAddress = Address.fromBase58(bridgeConstants.getBtcParams(), bridgeConstants.getOldFederationAddress());
+        }
```

Comment:
I prefer it the way it was before, you already checking for RSKIP199 activation in `getTransactionType` so just pass the old federation address value and we avoid passing `null` parameters

Reply:
The thing is the way it was before was the reason why test were failing because it was trying to get an Address from a null.